pkg/detector/library/advisory.go

```diff
@@ -45,11 +45,17 @@ func (s *Advisory) DetectVulnerabilities(pkgName string, pkgVer *semver.Version)
 			continue
 		}
 
+		var avdURL string
+		if strings.Contains(strings.ToLower(advisory.VulnerabilityID), "cve") {
+			avdURL = fmt.Sprintf("https://avd.aquasec.com/nvd/%s", strings.ToLower(advisory.VulnerabilityID))
+		}
+
```

Comment:
Is there any reason that we don't generate an AVD link in `FillInfo`?
https://github.com/aquasecurity/trivy/blob/793a1aa3c83f7a77355e454a12d034964bfa0088/pkg/vulnerability/vulnerability.go#L52-L83

Reply:
No reason - what's argument in favour?

Reply:
One argument I have against adding it is that in that case we have to solely rely on integration tests far more than we can on unit tests. AFAICT `FillInfo` cannot be easily unit tested and it isn't today.

Nothing against integration testing - just the shorter feedback loop unit testing is something I see enough of a reason to stick with it.

Reply:
`dbc` can be injected, so we can write unit tests now, right? If we change the AVD link a little, we have to fix all places now.

Reply:
OK. I did it here https://github.com/aquasecurity/trivy/pull/709/commits/934f993f7e0cf5cfd3c6313c8af42e6e2f9904ba